services/QuillLMS/spec/models/question_spec.rb

```diff
@@ -526,4 +526,105 @@
     end
   end
 
+  describe 'create_translation_mappings' do
+    subject {question.create_translation_mappings}
+
+    let(:question) { create(:question, data: {'instructions' => instructions})}
+    let(:instructions) { "do this question"}
+    let(:english_texts ) { question.english_texts }
+
+    it 'creates translation_mappings for focusPoints and incorrectSequences' do
+      expect(question).to receive(:create_data_translation_mappings).with(type: Question::FOCUS_POINTS)
+      expect(question).to receive(:create_data_translation_mappings).with(type: Question::INCORRECT_SEQUENCES)
+      subject
+    end
+
+    it 'creates translation_mappings for instructions' do
+      subject
+      instructions_mapping = question.translation_mappings.find_by(field_name: "instructions")
+      expect(instructions_mapping.text).to eq(instructions)
+    end
+  end
+
+  describe 'create_data_translation_mappings(type:)' do
+    subject {question.create_data_translation_mappings(type: )}
+
+    let(:type) { Question::FOCUS_POINTS }
+    let(:question) { create(:question)}
+    let(:translatable_data) { { 'focusPoints.uid1' => 'f1', 'focusPoints.uid2' => 'f2'}}
+
+    before do
+      allow(question).to receive(:translatable_data)
+        .with(type:)
+        .and_return(translatable_data)
```

Comment:
Stubbing the method that's under test here feels like it may be hiding bugs.  For instance, I flagged the hard-coding of `incorrectSequence` into the method in the model, which means I'm not sure that you can actually get the code as-written to produce the `translatable_data` shape that you're declaring in this spec.  If I'm right in that assumption, that failure is hidden by stubbing the call here.

This may actually be reasonable since you have an explicit set of specs below focused on `translatable_data` below, so this may not need to actually be "fixed", but again I wanted to call out that this felt risky to me.